butterfly-core/src/main/java/com/paypal/butterfly/core/ConfigurationImpl.java

```diff
@@ -24,7 +24,7 @@
     private boolean zipOutput = false;
     private boolean modifyOriginalFolder = true;
 
-    private static final Pattern propertyNameRegex = Pattern.compile("^[a-zA-Z\\._-]*$");
```

Comment:
Thanks @VietnameZe !! Notice the original regex also allowed dot, underscore and hyphen. Please, modify the new regex to still allow those characters.

Reply:
now it allow period, hyphen, underscore at beginning. What about these chars in mid string?

Reply:
Thanks Huy. Notice your branch seems to be out of sync, this PR has a conflict. Look how many commits it is bring in, and how many changed files it has. 

Also, your regex allows dot, underscore and hyphen as first character. First character must alphabetic only. I believe this is the best regex: `^[a-zA-Z][a-zA-Z0-9\._-]*$`. Notice in Java you would have to escape the slash, so it would be like this:

```java
Pattern.compile("^[a-zA-Z][a-zA-Z0-9\\._-]*$"); 
```

By the way, this is a good place to test regex: https://regex101.com/

Thanks.

Reply:
Hello @VietnameZe. Just double checking, will address the comment in this PR? Please let me know. Thanks.